src/main/java/com/microsoft/aad/msal4j/HttpHelper.java

```diff
@@ -6,32 +6,63 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import java.util.List;
-import java.util.Map;
+import java.util.*;
+
+import static com.microsoft.aad.msal4j.Constants.POINT_DELIMITER;
 
 class HttpHelper {
 
     private final static Logger log = LoggerFactory.getLogger(HttpHelper.class);
+    public static final String RETRY_AFTER_HEADER = "Retry-After";
+    public static final int RETRY_NUM = 2;
+    public static final int RETRY_DELAY_MS = 1000;
+
+    private static String getRequestThumbprint(RequestContext requestContext) {
+        StringBuilder sb = new StringBuilder();
+        sb.append(requestContext.clientId() + POINT_DELIMITER);
+        sb.append(requestContext.authority() + POINT_DELIMITER);
+
+        IApiParameters apiParameters = requestContext.apiParameters();
+
+        if (apiParameters instanceof SilentParameters) {
+            IAccount account = ((SilentParameters) apiParameters).account();
+            if (account != null) {
+                sb.append(account.homeAccountId() + POINT_DELIMITER);
+            }
+        }
+
+        Set<String> sortedScopes = new TreeSet<>(apiParameters.scopes());
+        sb.append(String.join(" ", sortedScopes));
 
-    static IHttpResponse executeHttpRequest(final HttpRequest httpRequest,
-                                            final RequestContext requestContext,
-                                            final ServiceBundle serviceBundle) {
+        return StringHelper.createSha256Hash(sb.toString());
+    }
+
+    static private boolean isRetriable(IHttpResponse httpResponse) {
```

Comment:
nit: typo retryable

Reply:
ok, will rename it, was not sure what it correct spelling